### PR TITLE
[1.4] Emit event when scroll top has been reached

### PR DESCRIPTION
**Pull request by alex-w0**
https://github.com/theomessin/vue-chat-scroll/issues/55

The new event is called v-chat-scroll-top-reached

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
   "author": "Theodore Messinezis",
   "license": "MIT",
   "peerDependencies": {
-    "vue": "^2.4.4"
+    "vue": "^2.6.11"
```
Comment (theomessin) [inline package.json:28]: No need to change all this.

File: src/directives/v-chat-scroll.js [hunk 2]
```diff
@@ -19,35 +19,44 @@ const scrollToBottom = (el, smooth) => {
...
       scrollToBottom(el, smooth);
-    })).observe(el, { childList: true, subtree: true });
+    }).observe(el, { childList: true, subtree: true });
   },
   inserted: (el, binding) => {
     const config = binding.value || {};
+
     scrollToBottom(el, config.smooth);
-  },
+  }
```
Comment (theomessin) [inline src/directives/v-chat-scroll.js:59]: No need to change all this.

File: src/directives/v-chat-scroll.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-*/
+ * @name VueJS vChatScroll (vue-chat-scroll)
+ * @description Monitors an element and scrolls to the bottom if a new child is added
+ * @author Theodore Messinezis <theo@theomessin.com>
+ * @file v-chat-scroll  directive definition
+ */
 
 const scrollToBottom = (el, smooth) => {
-  if (typeof el.scroll === "function") {
+  if (typeof el.scroll === 'function') {
```
Comment (theomessin) [inline src/directives/v-chat-scroll.js:9]: No need to change all this.

**Comment by theomessin** [comment]
Additionally, you need to add a section in `readme.md` documenting this feature.

**Comment by alex-w0** [comment]
Oh didn't notice that prettifier automatically refactored the code.
I will refactor the pull request today and let you know when it's done

**Comment by alex-w0** [comment]
@theomessin You can check it now again. thanks

**Comment by theomessin** [comment]
@alex-w0 why do you need `always` to be set to false?

**Comment by alex-w0** [comment]
> @alex-w0 why do you need `always` to be set to false?

If someone uses this event, then he will probably also add elements at the beginning. When always is now set to true, then it will be immediatly scrolled to the bottom, which in my opinion should not be, because the newest messages are at the top.

Or do you think we should remove that dependency?

**Comment by theomessin** [comment]
I'd rather let the user decide that for themselves. Maybe they have some other reason to want this event that we can't think of. Let's just add a warning in the docs saying that they need always if they intend to add items.

**Comment by alex-w0** [comment]
Alright, I've updated the pull request.

**Pull request merged**